modules/core/src/lib/layer.ts

```diff
@@ -745,8 +746,35 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
 
   /** (Internal) Sets the picking color at the specified index to null picking color. Used for multi-depth picking.
      This method may be overriden by layer implementations */
-  disablePickingIndex(objectIndex: number): void {
-    this._disablePickingIndex(objectIndex);
+  disablePickingIndex(objectIndex: number) {
+    const data = this.props.data as LayerData<any>;
+    if (!('attributes' in data)) {
+      this._disablePickingIndex(objectIndex);
+      return;
+    }
+
+    // @ts-ignore (TS2531) this method is only called internally with attributeManager defined
+    const {pickingColors, instancePickingColors} = this.getAttributeManager().attributes;
+    const colors = pickingColors || instancePickingColors;
+    const externalColorAttribute =
+      colors && data.attributes && (data.attributes[colors.id] as BinaryAttribute);
+    if (externalColorAttribute && externalColorAttribute.value) {
+      const values = externalColorAttribute.value;
+      const objectColor = this.encodePickingColor(objectIndex);
+      for (let index = 0; index < data.length; index++) {
+        const i = colors.getVertexOffset(index);
+        if (
+          values[i] === objectColor[0] &&
+          values[i + 1] === objectColor[1] &&
+          values[i + 2] === objectColor[2]
+        ) {
+          this._disablePickingIndex(index);
+          break;
```

Comment:
This break will prevent disabling picking colors correctly for multi polygons or multi line strings -- every match needs to disabled, not just the first one.

Reply:
I see, sorry about that. Could you kill the line and make sure this PR fixes the original issue before we merge? I think the only thing missing is unit tests, but I can add that in a follow up PR.

Reply:
Removed + verified that it works as I'd expect now!